src/main/java/duke/DukeListener.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.command.CommandExit;
+
+
+/**
+ * Waits for and processes a User Input
+ */
+public class DukeListener {
+    private final Scanner sc;
```

Comment:
No harm calling it `scanner` too. Would be clearer in parts of code that uses this variable.

Reply:
You mention a very good point!